src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -510,6 +511,25 @@ private AbstractFrame getFieldWithoutThis(DetailAST ast, int parentType) {
         return frame;
     }
 
+    /**
+     * Return whether ast is in a COMPACT_CTOR_DEF.
+     *
+     * @param ast The token to check
+     * @return true if ast is in a COMPACT_CTOR_DEF, false otherwise
+     */
+    private static boolean isInCompactConstructor(DetailAST ast) {
+        boolean isInCompactCtor = false;
+        DetailAST tempAst = ast;
```

Comment:
This variable can be initialized with `ast.getParent()`, so the loop simplifies to
```java
 DetailAST tempAst = ast.getParent();
 while (tempAst != null) {
   if (tempAst.getType() == TokenTypes.COMPACT_CTOR_DEF) {
   ...
```
also, `tempAst` can be renamed to `parent` for readability.

Reply:
fixed